coderd/templateversions_test.go

```diff
@@ -720,22 +816,45 @@ func TestPaginatedTemplateVersions(t *testing.T) {
 	_ = coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
 	template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID)
 
+	// This test takes longer than a long time.
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong*2)
+	defer cancel()
+
 	// Populate database with template versions.
 	total := 9
+	eg, egCtx := errgroup.WithContext(ctx)
+	templateVersionIDs := make([]uuid.UUID, total)
 	for i := 0; i < total; i++ {
-		data, err := echo.Tar(nil)
-		require.NoError(t, err)
-		file, err := client.Upload(context.Background(), codersdk.ContentTypeTar, data)
-		require.NoError(t, err)
-		templateVersion, err := client.CreateTemplateVersion(ctx, user.OrganizationID, codersdk.CreateTemplateVersionRequest{
-			TemplateID:    template.ID,
-			StorageSource: file.Hash,
-			StorageMethod: codersdk.ProvisionerStorageMethodFile,
-			Provisioner:   codersdk.ProvisionerTypeEcho,
+		i := i
+		eg.Go(func() error {
+			data, err := echo.Tar(nil)
+			if err != nil {
+				return err
+			}
+			file, err := client.Upload(egCtx, codersdk.ContentTypeTar, data)
+			if err != nil {
+				return err
+			}
+			templateVersion, err := client.CreateTemplateVersion(egCtx, user.OrganizationID, codersdk.CreateTemplateVersionRequest{
+				TemplateID:    template.ID,
+				StorageSource: file.Hash,
+				StorageMethod: codersdk.ProvisionerStorageMethodFile,
+				Provisioner:   codersdk.ProvisionerTypeEcho,
+			})
+			if err != nil {
+				return err
+			}
+
+			templateVersionIDs[i] = templateVersion.ID
+
+			return nil
```

Comment:
This is the sort of thing where it would be really useful to have a mock database and just return some predefined fixture.

Reply:
Very much agreed!